ArduPlane/soaring.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Plane.h"
+
+/*
+*  ArduSoar support function
+*
+*  Peter Braswell, Samuel Tabor, Andrey Kolobov, and Iain Guilliard
+*/
+void Plane::update_soaring() {
+    
+    if (!g2.soaring_controller.is_active()) {
+        return;
+    }
+    
+    g2.soaring_controller.update_vario();
+
+    // Check for throttle suppression change.
+    switch (control_mode){
+    case AUTO:
+        g2.soaring_controller.suppress_throttle();
+        break;
+    case FLY_BY_WIRE_B:
+    case CRUISE:
```

Comment:
The original implementation activated LOITER  from AUTO or FBWB mode. LOITER exited back to AUTO if activated from AUTO, or to RTL if activated from FBWB. Throttle was activated if SOAR_MIN_ALT is breached in either case. I agree we should stick to this, the question is, what should CRUISE do